src/battle_main.c

```diff
@@ -3344,6 +3344,19 @@ const u8* FaintClearSetData(u32 battler)
     gBattleStruct->palaceFlags &= ~(1u << battler);
     gBattleStruct->boosterEnergyActivates &= ~(1u << battler);
 
+    if (gBattleStruct->commanderActive[battler] != SPECIES_NONE)
+    {
+        u32 partner = BATTLE_PARTNER(battler);
+        gBattleStruct->commanderActive[battler] = SPECIES_NONE;
+        if (IsBattlerAlive(partner))
+        {
+            gBattleStruct->commandingDondozo &= ~(1u << partner);
+            gStatuses3[partner] &= ~STATUS3_COMMANDER;
+            BtlController_EmitSpriteInvisibility(partner, BUFFER_A, FALSE);
+            MarkBattlerForControllerExec(partner);
+        }
+    }
+
```

Comment:
Might be worth considering doing this in MoveEnd if "(Commander) Tatsugiri is not damaged by a double target move if Dondozo faints" still fails after my suggestion (by trying to "disable" Commander from both sides at the end of the move)

Reply:
To many tests fail when I try to reset it in `moveened`

Reply:
```suggestion
    if (gBattleStruct->commanderActive[battler] != SPECIES_NONE)
    {
        u32 partner = BATTLE_PARTNER(battler);
        if (IsBattlerAlive(partner))
        {
            BtlController_EmitSpriteInvisibility(partner, BUFFER_A, FALSE);
            MarkBattlerForControllerExec(partner);
        }
    }

```

Then add this to `MOVEEND_CLEAR_BITS`:

```
for (i = 0; i < gBattlersCount; i++)
{
    if (gBattleStruct->commanderActive[i] != SPECIES_NONE && !IsBattlerAlive(i))
    {
         u32 partner = BATTLE_PARTNER(i);
         gBattleStruct->commanderActive[i] = SPECIES_NONE;
         if (IsBattlerAlive(partner))
         {
              gBattleStruct->commandingDondozo &= ~(1u << partner);
              gStatuses3[partner] &= ~STATUS3_COMMANDER;
         }
     }
}
```

Looks a little weird because the Tatsugiri is visible and just avoids the attack, but it should be good when Simultaneous HP reduction is implemented

Reply:
Yep, that works, thx!

I think with Simultaneous HP reduction we wouldn't even need this exception but for now it is a good solution imo. I had tunnel vision on this for sure.